### PR TITLE
class alias tags in html generated from markdown

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: tools/gulp/tasks/docs.ts [hunk 1]
```diff
@@ -36,6 +55,15 @@ gulp.task('markdown-docs', () => {
         }
       }))
       .pipe(transform(transformMarkdownFiles))
+      .pipe(dom(function () {
+        MARKDOWN_TAGS_TO_CLASS_ALIAS.forEach((tag) => {
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: Move this into its own function like `fixMarkdownDocLinks`?

File: tools/gulp/tasks/docs.ts [hunk 2]
```diff
@@ -36,6 +55,15 @@ gulp.task('markdown-docs', () => {
...
+          Array.prototype.slice.call(this.querySelectorAll(tag)).forEach((el: any) => {
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: Would it work to do 
```
for (let element of this.querySelectorAll(tag))
``` 
?

File: tools/gulp/tasks/docs.ts [hunk 3]
```diff
@@ -19,6 +20,24 @@ const EXAMPLE_PATTERN = /<!--\W*example\(([^)]+)\)\W*-->/g;
...
+    'h2',
+    'h3',
+    'h4',
+    'h5',
+    'li',
+    'ol',
+    'p',
+    'table',
+    'tbody',
+    'td',
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: `tr`?

File: tools/gulp/tasks/docs.ts [hunk 4]
```diff
@@ -19,6 +20,24 @@ const EXAMPLE_PATTERN = /<!--\W*example\(([^)]+)\)\W*-->/g;
...
+// for styling purposes.
+const MARKDOWN_TAGS_TO_CLASS_ALIAS = [
+    'a',
+    'h1',
+    'h2',
+    'h3',
+    'h4',
+    'h5',
+    'li',
+    'ol',
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: `ul`?

File: tools/gulp/tasks/docs.ts [hunk 5]
```diff
@@ -95,3 +117,20 @@ function fixMarkdownDocLinks(link: string, filePath: string): string {
...
   return `guide/${baseName}`;
 }
+
+/**
+ * Returns a function to be called with an HTML document as its context that aliases HTML tags by
+ * adding a class consisting of a prefix + the tag name.
+ * @param classPrefix The prefix to use for the alias class.
+ */
+function createTagNameAliaser(classPrefix: string) {
+  return function() {
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: nit: can be an arrow function

File: tools/gulp/tasks/docs.ts [hunk 6]
```diff
@@ -95,3 +117,20 @@ function fixMarkdownDocLinks(link: string, filePath: string): string {
...
+    MARKDOWN_TAGS_TO_CLASS_ALIAS.forEach((tag) => {
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: nit: parens aren't needed

File: tools/gulp/tasks/docs.ts (hunk 5 above)
Comment (mmalerba) [inline tools/gulp/tasks/docs.ts]: It actually won't get the correct context if I use an arrow fn

**Pull request merged**